server/server.js

```diff
@@ -21,10 +26,62 @@ app.get('/', function(request, response) {
 });
 
 io.on('connection', (socket) => {
-  socket.emit('message', 'hello from server');
+  //Reply to application after receiving hello message
+  socket.on('hello', (msg)=> {
+      console.log("Message received: " + msg);
+      socket.emit('reply', "hello from the server side");
+  });
   console.log("Client connected");
 });
 
 server.listen(app.get('port'), function() {
   console.log("Spots server is running on http://localhost:%s", app.get('port'));
 });
+
+//--------------------------------------------------------------------------------------------
+//Functions
+//--------------------------------------------------------------------------------------------
+
+//Registers User to database
+//Paramters: (int, string, string, string, int, boolean, boolean, string?, permitType,
+//  string?, string?, string?, string?)
+function registerUser(userId, first, last, email, phone, gotPermit, expired, expDate,
+  permittype, make, model, color, plate){
```

Comment:
This is a long list of arguments. Let's consider making a `userProfile` class to hold all this data in a future PR.